docs/api/js/utils.md

```diff
@@ -1,68 +1,772 @@
+- `signature`: ethers signature as `SignatureLike` object. 
+
+```ts
+async function isEIP1271SignatureCorrect(
+    provider: Provider,
+    address: string,
+    msgHash: string,
+    signature: SignatureLike
+): Promise<boolean> {
+    const accountContract = new ethers.Contract(address, IERC1271, provider);
+
+    // This line may throw an exception if the contract does not implement the EIP1271 correctly.
+    // But it may also throw an exception in case the internet connection is lost.
+    // It is the caller's responsibility to handle the exception.
+    const result = await accountContract.isValidSignature(msgHash, signature);
+
+    return result == EIP1271_MAGIC_VALUE;
+}
+```
+
+### `isETH`
+
+Returns true if token represents ETH on L1 or L2.
+
+#### Inputs
+
+- `token`: `Address` object.
+
+```ts
+export function isETH(token: Address) {
+    return token.toLowerCase() == ETH_ADDRESS || token.toLowerCase() == L2_ETH_TOKEN_ADDRESS;
+}
+```
+
+### `isMessageSignatureCorrect`
+
+Returns true if account abstraction EIP712 signature is correct.
+
+#### Inputs
+
+- `provider`: `Provider` object.
+- `address`: sender address as string.
+- `message`: string or Bytes object representation of the message. 
+- `signature`: ethers signature as `SignatureLike` object.
+
+```ts
+export async function isMessageSignatureCorrect(
+    provider: Provider,
+    address: string,
+    message: ethers.Bytes | string,
+    signature: SignatureLike
+): Promise<boolean> {
+    const msgHash = ethers.utils.hashMessage(message);
+    return await isSignatureCorrect(provider, address, msgHash, signature);
+}
+```
+
+### `isSignatureCorrect`
+
+Called from [`isMessageSignatureCorrect`](#ismessagesignaturecorrect) and [`isTypedDataSignatureCorrect`](#istypeddatasignaturecorrect). Returns true if account abstraction EIP712 signature is correct.
+
+#### Inputs
+
+- `provider`: `Provider` object.
+- `address`: sender address as string.
+- `msgHash`: hash of the message as string. 
+- `signature`: ethers signature as `SignatureLike` object.
+
+```ts
+async function isSignatureCorrect(
+    provider: Provider,
+    address: string,
+    msgHash: string,
+    signature: SignatureLike
+): Promise<boolean> {
+    let isContractAccount = false;
+
+    const code = await provider.getCode(address);
+    isContractAccount = ethers.utils.arrayify(code).length != 0;
+
+    if (!isContractAccount) {
+        return isECDSASignatureCorrect(address, msgHash, signature);
+    } else {
+        return await isEIP1271SignatureCorrect(provider, address, msgHash, signature);
+    }
+}
+```
+
+### `isTypedDataSignatureCorrect`
+
+Returns true if account abstraction EIP712 signature is correct.
+
+#### Inputs
+
+- `provider`: Provider object.
+- `address`: sender address as string.
+- `domain`: `TypedDataDomain` object from `@ethersproject/abstract-signer`.
+- `types`: map of records pointing to array of data as `Record<string, Array<TypedDataField>>` ??. 
```

Comment:
map of recording pointing from the field's name to its type

Reply:
done